plugins/Denizen/scripts/cooking/allergies.dsc

```diff
@@ -0,0 +1,77 @@
+allergies:
+    type: world
+    debug: false
+    events:
+        on player dies:
+            - if <util.random_chance[12]>:
+                - define allergy <script[allergy_info].data_key[valid_allergies].keys.random>
+                - define effect <script[allergy_info].data_key[valid_effects].keys.random>
+                - flag player allergic:<[allergy]>
+                - flag player allergy_effect:<[effect]>
+                - stop
+            - flag player allergic:!
+            - flag player allergy_effect:!
+
+        on player consumes item flagged:allergic:
```

Comment:
```suggestion
        on entity_flagged:allergic consumes item:
```
The `flagged` switch here is technically correct however it can be simplified by putting the `entity_flagged` instead.